packages/feedback/src/core/sendFeedback.ts

```diff
@@ -1,100 +1,65 @@
-import { createAttachmentEnvelope, createEventEnvelope, getClient, withScope } from '@sentry/core';
-import type { FeedbackEvent, SendFeedback, SendFeedbackParams } from '@sentry/types';
+import { captureFeedback } from '@sentry/core';
+import { getClient } from '@sentry/core';
+import type { EventHint, SendFeedback, SendFeedbackParams, TransportMakeRequestResponse } from '@sentry/types';
+import type { Event } from '@sentry/types';
 import { getLocationHref } from '@sentry/utils';
-import { FEEDBACK_API_SOURCE, FEEDBACK_WIDGET_SOURCE } from '../constants';
-import { prepareFeedbackEvent } from '../util/prepareFeedbackEvent';
+import { FEEDBACK_API_SOURCE } from '../constants';
 
 /**
  * Public API to send a Feedback item to Sentry
  */
 export const sendFeedback: SendFeedback = (
-  { name, email, message, attachments, source = FEEDBACK_API_SOURCE, url = getLocationHref() }: SendFeedbackParams,
-  { includeReplay = true } = {},
-) => {
-  if (!message) {
+  options: SendFeedbackParams,
+  hint: EventHint & { includeReplay?: boolean } = { includeReplay: true },
+): Promise<string> => {
+  if (!options.message) {
     throw new Error('Unable to submit feedback with empty message');
   }
 
+  // We want to wait for the feedback to be sent (or not)
   const client = getClient();
-  const transport = client && client.getTransport();
-  const dsn = client && client.getDsn();
 
-  if (!client || !transport || !dsn) {
-    throw new Error('Invalid Sentry client');
+  if (!client) {
+    throw new Error('No client setup, cannot send feedback.');
   }
 
-  const baseEvent: FeedbackEvent = {
-    contexts: {
-      feedback: {
-        contact_email: email,
-        name,
-        message,
-        url,
-        source,
-      },
+  const eventId = captureFeedback(
+    {
+      source: FEEDBACK_API_SOURCE,
+      url: getLocationHref(),
+      ...options,
     },
-    type: 'feedback',
-  };
+    hint,
+  );
 
-  return withScope(async scope => {
-    // No use for breadcrumbs in feedback
-    scope.clearBreadcrumbs();
+  // We want to wait for the feedback to be sent (or not)
+  return new Promise<string>((resolve, reject) => {
+    // After 5s, we want to clear anyhow
+    const timeout = setTimeout(() => reject('Unable to determine if Feedback was correctly sent.'), 5_000);
 
-    if ([FEEDBACK_API_SOURCE, FEEDBACK_WIDGET_SOURCE].includes(String(source))) {
-      scope.setLevel('info');
-    }
-
-    const feedbackEvent = await prepareFeedbackEvent({
-      scope,
-      client,
-      event: baseEvent,
-    });
-
-    if (client.emit) {
-      client.emit('beforeSendFeedback', feedbackEvent, { includeReplay: Boolean(includeReplay) });
```

Comment:
I think we still want to have this beforeSendFeedback hook. This should a) start to capture any buffered replay b) add a breadcrumb to the replay

Reply:
Long term it'd be better to move into the Actor::onClick event, so we get more replay before they start typing, but that should be a followup because of the quota impact

Reply:
We do still emit this, but in `captureFeedback` ;)